pkg/devfile/adapters/kubernetes/component/adapter_test.go

```diff
@@ -555,6 +569,148 @@ func TestAdapterDelete(t *testing.T) {
 	}
 }
 
+func TestCreateDockerConfigSecretBytes(t *testing.T) {
+
+	testConfigJsonString := "{ \"auths\" : { \"https://index.docker.io/v1/\": { \"auth\": \"test-auth-token\", \"email\": \"test-email\"} },\"HttpHeaders\": {	\"User-Agent\": \"Docker-Client/19.03.8 (darwin)\"},\"experimental\": \"disabled\"}"
+	testFs := NewFilesystem()
+	testFilename := "test-config-json"
+	testConfigJsonBytes := []byte(testConfigJsonString)
+
+	defer testFs.Remove(testFilename)
+
+	afero.WriteFile(testFs, testFilename, []byte(testConfigJsonString), 0777)
+
+	fkclient, _ := kclient.FakeNew()
+	testAdapter := Adapter{
+		Client: *fkclient,
+	}
+
+	want := testConfigJsonBytes
+	got, err := testAdapter.createDockerConfigSecretBytes(testFilename)
+	if err != nil {
+		t.Error(err)
+		t.Errorf("failed to retrieve dockerconfig secret bytes")
+	}
+
+	if !reflect.DeepEqual(want, got) {
+		t.Errorf("secret bytes don't match")
+	}
+}
+
+func TestCreateDockerConfigSecret(t *testing.T) {
+	testConfigJsonString := "{ \"auths\" : { \"https://index.docker.io/v1/\": { \"auth\": \"test-auth-token\", \"email\": \"test-email\"} },\"HttpHeaders\": {	\"User-Agent\": \"Docker-Client/19.03.8 (darwin)\"},\"experimental\": \"disabled\"}"
+	testDockerConfigData := []byte(testConfigJsonString)
+	testSecretName := "test-secret"
+	testNs := "test-namespace"
+
+	testNamespacedName := types.NamespacedName{
+		Name:      testSecretName,
+		Namespace: testNs,
+	}
+
+	testSecret := &corev1.Secret{
+
+		TypeMeta:   TypeMeta("Secret", "v1"),
+		ObjectMeta: SecretObjectMeta(testNamespacedName),
+		Type:       corev1.SecretTypeDockerConfigJson,
+		Data: map[string][]byte{
+			corev1.DockerConfigJsonKey: testDockerConfigData,
+		},
+	}
+
+	testSecretData, err := runtimeUnstructured.DefaultUnstructuredConverter.ToUnstructured(testSecret)
+	if err != nil {
+		t.Error(err)
+		t.Errorf("error making map")
+	}
+
+	testSecretBytes, err := json.Marshal(testSecretData)
+	if err != nil {
+		t.Error(err)
+		t.Errorf("error while marshalling")
+	}
+
+	var testSecretUnstructured *unstructured.Unstructured
+	if err := json.Unmarshal(testSecretBytes, &testSecretUnstructured); err != nil {
+		t.Error(err)
+		t.Errorf("error unmarshalling into unstructured")
+	}
+
+	want := testSecretUnstructured
+
+	fkclient, _ := kclient.FakeNew()
+
+	scheme := runtime.NewScheme()
+	fkclient.DynamicClient = dynamicfakeclient.NewSimpleDynamicClient(scheme)
+
+	if err != nil {
+		t.Errorf("error generating fake dynamic client")
+	}
+
+	testAdapter := Adapter{
+		Client: *fkclient,
+	}
+
+	got, err := testAdapter.createDockerConfigSecret(testSecretName, testNs, testDockerConfigData)
+
+	if err != nil {
+		t.Error(err)
+		t.Errorf("failed to get secret")
+	}
+
+	if !reflect.DeepEqual(got, want) {
```

Comment:
Use `cmp.Diff` instead.    The output from cmp.Diff can be reported and used for troubleshooting.  It return "" if the two objects are identical.

Reply:
Done